tests/test_client_request.py

```diff
@@ -1230,3 +1230,9 @@ def test_loose_cookies_types(loop: Any) -> None:
 def test_gen_default_accept_encoding(has_brotli: Any, expected: Any) -> None:
     with mock.patch("aiohttp.client_reqrep.HAS_BROTLI", has_brotli):
         assert _gen_default_accept_encoding() == expected
+
+
+async def test_certificate(loop: Any, conn: Any) -> None:
+    req = ClientRequest("get", URL("https://www.python.org"), loop=loop)
```

Comment:
Instead of reaching out to the network, it's best to use a local test server, extracting it's certificate.

Reply:
Not sure about this. A real server allows one to test in various conditions that are not easy to recreate in a controlled environment, such as premature loss of transport or connection, redirects, proxy servers, and is closest to the experience of the end user. On the other hand, if the connection is closed immediately due to a transient issue, this test will fail. As many tests depend on a real server, this should not be much of an issue.

Reply:
> As many tests depend on a real server, this should not be much of an issue.

What tests? I'm not sure our test suite needs an external internet connection at all.
Adding a test which relies on an internet connection and a working server at python.org is not a good way to go. We have many packagers etc. who run our tests as well, so it's not just us this would affect.

Reply:
Much of the code in the test was copied from another test, such as https://github.com/aio-libs/aiohttp/blob/b0a679f9247fce4e3600d7ef5ea796b9d09b6862/tests/test_client_request.py#L562 There are several such tests in the suite.